core/src/main/java/io/jenkins/pluginhealth/scoring/probes/HasUnreleasedProductionChangesProbe.java

```diff
@@ -0,0 +1,202 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.jenkins.pluginhealth.scoring.probes;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.*;
+import java.util.regex.Matcher;
+
+import io.jenkins.pluginhealth.scoring.model.Plugin;
+import io.jenkins.pluginhealth.scoring.model.ProbeResult;
+
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.LogCommand;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.diff.DiffEntry;
+import org.eclipse.jgit.diff.DiffFormatter;
+import org.eclipse.jgit.errors.CorruptObjectException;
+import org.eclipse.jgit.errors.IncorrectObjectTypeException;
+import org.eclipse.jgit.errors.MissingObjectException;
+import org.eclipse.jgit.lib.ObjectId;
+import org.eclipse.jgit.lib.ObjectReader;
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.eclipse.jgit.revwalk.RevTree;
+import org.eclipse.jgit.revwalk.RevWalk;
+import org.eclipse.jgit.treewalk.CanonicalTreeParser;
+import org.eclipse.jgit.treewalk.TreeWalk;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+/**
+ * Using the analysis done by {@link SCMLinkValidationProbe},
+ * this probe will find number of unreleased commits in a repository
+ */
+@Component
+@Order(value = HasUnreleasedProductionChangesProbe.ORDER)
+public class HasUnreleasedProductionChangesProbe  extends Probe {
+    private static final Logger LOGGER = LoggerFactory.getLogger(HasUnreleasedProductionChangesProbe.class);
+
+    public static final int ORDER = LastCommitDateProbe.ORDER + 100;
+    public static final String KEY = "unreleased-production-changes";
+
+    @Override
+    public ProbeResult doApply(Plugin plugin, ProbeContext context) {
+        List<String> productionPathsToCheckForCommits = new ArrayList<>();
+        Set<String> filesModifiedAfterRelease  = new HashSet<>();
+
+        final Matcher matcher = SCMLinkValidationProbe.GH_PATTERN.matcher(plugin.getScm());
+        if (!matcher.find()) {
+            return ProbeResult.failure(key(), "The SCM link is not valid");
+        }
+        final String folder = matcher.group("folder");
+
+        try (Git git = Git.init().setDirectory(context.getScmRepository().toFile()).call()) {
+            final LogCommand logCommand = git.log();
+
+            if (folder != null) {
+                productionPathsToCheckForCommits.add(folder + "/pom.xml");
+                productionPathsToCheckForCommits.add(folder + "/src/main");
+            }
+            else {
+                productionPathsToCheckForCommits.add("pom.xml");
```

Comment:
It is also required when you have a folder as the parent pom can be managing dependencies which you need to track.